### PR TITLE
Fix Dockerfiles and docker images

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR fixes several docker image related issues

- [x] Remove support for dependency image builds from the framework 
- [ ] Separate agent image into base autonomy image and runtime image

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [x] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

### CODE REVIEW

File: autonomy/cli/deploy.py [hunk 1]
```diff
@@ -266,13 +266,6 @@ def run_deployment_from_token(  # pylint: disable=too-many-arguments, too-many-l
                 version=DEFAULT_IMAGE_VERSION,
                 push=False,
             )
-            build_image(
```
Comment (DavidMinarsch) [inline autonomy/cli/deploy.py:269]: So this was responsible for building deps right?

Comment (angrybayblade) [reply autonomy/cli/deploy.py:269]: Yes

**Review by DavidMinarsch — COMMENTED** [review]
Why did you revert the registry to an old state? Please fix it!

**Comment by angrybayblade** [comment]
> Why did you revert the registry to an old state? Please fix it!

So as you kknow, aleksander introduced the mainnet snapshot utility on friday but I did not had enough time to discuss this change with him so I faced some issues while running the local deployment setup for the workshop. So reverted to the last commit where it was working for me, but then I forgot to checkout to head 😓

File: .github/workflows/main_workflow.yml [hunk 2]
```diff
@@ -213,17 +213,11 @@ jobs:
...
-          cd hello_world
-          autonomy build-images
-          autonomy build-images --dev
-          autonomy build-images --dependencies
-          autonomy build-images --dependencies --version 1.0.0
-          autonomy deploy build ../deployments/keys/hardhat_keys.json
-
-          cd ..
+          skaffold build --push=false
+          skaffold build -p release --push=false
```
Comment (DavidMinarsch) [inline .github/workflows/main_workflow.yml:218]: These builds take a long time...so we should at least build with cache no? https://github.com/valory-xyz/open-autonomy/runs/8025408228?check_suite_focus=true 

Which means we need to have a specific scaffold flag just for the CI

File: .github/workflows/main_workflow.yml [hunk 3]
```diff
@@ -213,17 +213,11 @@ jobs:
...
+          docker build deployments/Dockerfiles/tendermint/ -t valory/open-autonomy-tendermint:0.1.0 --file deployments/Dockerfiles/tendermint/Dockerfile
+          docker build deployments/Dockerfiles/tendermint/ -t valory/open-autonomy-tendermint:1.0.0 --file deployments/Dockerfiles/tendermint/Dockerfile
```
Comment (DavidMinarsch) [inline .github/workflows/main_workflow.yml:220]: why again?

Comment (DavidMinarsch) [reply .github/workflows/main_workflow.yml:220]: IT's already in the skaffold file

File: docs/upgrading.md [hunk 4]
```diff
@@ -11,6 +11,7 @@ Multiple backwards incompatible changes:
 
 - The service config no longer accepts the `network` key on the first YAML page. The network can now be defined via the package overrides.
 - Dependency specifications are now checked against imports for all packages, this means initially packages might need modifying to reference/unreference missing/irrelevant dependencies.
+- The global configuration file for the `aea`/`autonomy` CLI has a breaking change. Please remove `~/.aea/cli_config.yaml` and rerun `autonomy init --remote`.
```
Comment (DavidMinarsch) [inline docs/upgrading.md:14]: @angrybayblade you need to describe what commands changed between `0.1.6` and `0.2.0`!

File: autonomy/cli/deploy.py [hunk 5]
```diff
@@ -257,22 +246,9 @@ def run_deployment_from_token(  # pylint: disable=too-many-arguments, too-many-l
...
-                skaffold_dir=DATA_DIR / DOCKERFILES,
-                version=DEFAULT_IMAGE_VERSION,
-                push=False,
-            )
-            build_image(
-                agent=service.agent,
-                profile=ImageProfiles.DEPENDENCIES,
-                skaffold_dir=DATA_DIR / DOCKERFILES,
-                version=DEFAULT_IMAGE_VERSION,
-                push=False,
```
Comment (dvilelaf) [inline autonomy/cli/deploy.py:274]: Are these options now removed or do they live somewhere else?

File: tests/helpers/deployments/test_deployments.py [hunk 6]
```diff
@@ -62,7 +62,7 @@
 description: Description
 aea_version: ">=1.0.0, <2.0.0"
 license: Apache-2.0
-agent: "valory/oracle:0.1.0"
+agent: valory/oracle:0.1.0:bafybeihurloujnbugvvrv5xegyrdnsgl6z6at5xilq7f5ynjdekijapt6q
```
Comment (DavidMinarsch) [inline tests/helpers/deployments/test_deployments.py:65]: This hash is not a current version...what is it pointing to? It should always match the version in packages/hashes.csv

File: .github/workflows/main_workflow.yml (hunk 2 above)
Comment (angrybayblade) [reply .github/workflows/main_workflow.yml:218]: We just need it until the release, once we perform the release and publish these images we'll just pull them in the CI.

File: .github/workflows/main_workflow.yml (hunk 3 above)
Comment (angrybayblade) [reply .github/workflows/main_workflow.yml:220]: We're tagging `latest` in the skaffold, and we need some other tags. Again just need to do this until the release, then we can just tag the existing image.

File: tests/helpers/deployments/test_deployments.py (hunk 6 above)
Comment (angrybayblade) [reply tests/helpers/deployments/test_deployments.py:65]: This is just a dummy config, we just use it to test if the service loader and deployment generators work or not.

**Review by dvilelaf — COMMENTED** [review]
There's a lot here outside of my knowledge, but looks good. I've left a question.

File: docs/upgrading.md (hunk 4 above)
Comment (angrybayblade) [reply docs/upgrading.md:14]: I'll do this in the release PR

File: autonomy/cli/deploy.py (hunk 5 above)
Comment (angrybayblade) [reply autonomy/cli/deploy.py:274]: Removed

File: tests/helpers/deployments/test_deployments.py (hunk 6 above)
Comment (DavidMinarsch) [reply tests/helpers/deployments/test_deployments.py:65]: Ok

File: docs/upgrading.md (hunk 4 above)
Comment (DavidMinarsch) [reply docs/upgrading.md:14]: Ok - in release PR both for the previous and current changes :)

File: .github/workflows/main_workflow.yml (hunk 2 above)
Comment (DavidMinarsch) [reply .github/workflows/main_workflow.yml:218]: Good point

Comment (DavidMinarsch) [reply .github/workflows/main_workflow.yml:218]: Please open an issue so we don't forget

Comment (angrybayblade) [reply .github/workflows/main_workflow.yml:218]: https://github.com/valory-xyz/open-autonomy/issues/1234

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**